packages/web/src/pages/upload-page/fields/RemixModalField.tsx

```diff
@@ -0,0 +1,144 @@
+import type { Track, UserMetadata, Nullable, ID } from '@audius/common'
+import {
+  getPathFromTrackUrl,
+  useGetTrackByPermalink,
+  SquareSizes,
+  accountSelectors
+} from '@audius/common'
+import { useField } from 'formik'
+import { useSelector } from 'react-redux'
+
+import { ReactComponent as IconRemix } from 'assets/img/iconRemixGray.svg'
+import {
+  InputV2,
+  InputV2Size,
+  InputV2Variant
+} from 'components/data-entry/InputV2'
+import { Divider } from 'components/divider'
+import DynamicImage from 'components/dynamic-image/DynamicImage'
+import UserBadges from 'components/user-badges/UserBadges'
+import { useTrackCoverArt } from 'hooks/useTrackCoverArt'
+import { withNullGuard } from 'utils/withNullGuard'
+
+import { ModalField } from './ModalField'
+import styles from './RemixModalField.module.css'
+import { ToggleRowField } from './ToggleRowField'
+
+const { getUserId } = accountSelectors
+
+const messages = {
+  title: 'Remix Settings',
+  description:
+    'Mark your music as a remix, tag the original track, and customize remix settings.',
+  by: 'by',
+  hideRemix: {
+    header: 'Hide Remixes of This Track',
+    description:
+      'Enable this option if you want to prevent remixes of your track by other artists from appearing on your track page.'
+  },
+  remixOf: {
+    header: 'Identify as Remix',
+    description:
+      "Paste the original Audius track link if yours is a remix. Your remix will typically appear on the original track's page.",
+    linkLabel: 'Link to Remix'
+  }
+}
+
+export type RemixOfField = Nullable<{ tracks: { parent_track_id: ID }[] }>
+
+const HIDE_REMIX_FIELD_NAME = 'hide_remixes'
+const REMIX_OF_FIELD_NAME = 'remix_of'
+const REMIX_LINK_FIELD_NAME = 'remix_of_link'
+
+export const RemixModalField = () => {
+  const [linkField] = useField(REMIX_LINK_FIELD_NAME)
+  const debugLink = linkField.value?.replace(
```

Comment:
we want dis?

Reply:
oh whoops yea doesn't work with localhost but shoudn't check this in :)